labtool2.0/src/components/pages/ReviewStudent.js

```diff
@@ -43,12 +50,43 @@ export class ReviewStudent extends Component {
     }
   }
 
+  toggleCheckbox = name => async e => {
+    this.props.toggleCheck(name)
+  }
+
+  copyChecklistOutput = async e => {
+    e.preventDefault()
+    this.reviewPointsRef.current.inputRef.value = e.target.points.value
+    /* The below line is as hacky as it is because functional elements cannot directly have refs.
+    * This abomination somehow accesses a textarea that is a child of a div that holds the ref.
+    */
+    this.reviewTextRef.current.children[0].children.comment.value = e.target.text.value
+  }
+
   render() {
     //this.props.ownProps.studentInstance is a string, therefore casting to number.
     const studentData = this.props.courseData.data.filter(dataArray => dataArray.id === Number(this.props.ownProps.studentInstance))
     //this.props.weekNumber is a string, therefore casting to number.
     const weekData = studentData[0].weeks.filter(theWeek => theWeek.weekNumber === Number(this.props.ownProps.weekNumber))
     const checkList = this.props.selectedInstance.checklists.find(checkl => checkl.week == this.props.ownProps.weekNumber)
+    let checklistOutput = ''
+    let checklistPoints = 0
+    if (checkList) {
+      Object.keys(checkList.list).forEach(cl => {
+        checkList.list[cl].forEach(row => {
+          const addition = this.props.weekReview.checks[row.name] ? row.textWhenOn : row.textWhenOff
+          if (addition) checklistOutput += addition + '\n\n'
+          if (this.props.weekReview.checks[row.name]) {
+            checklistPoints += row.points
+          }
+        })
+      })
+      if (checklistPoints < 0) {
+        checklistPoints = 0
+      } else if (checklistPoints > this.props.selectedInstance.weekMaxPoints) {
```

Comment:
I'm not sure if it's necessary to restrict the max points for the checklist. Usually the checklist points are assigned so that they cannot exceed the weekly maximum. In other words, there's only one way to get the maximum points: you have to meet all the criteria.

Now, if you make an error while making the checklist and the max possible points exceed the weekly max, you might not notice that something's wrong. 

But this should probably be checked when creating the checklist.